napari/layers/base/base.py

```diff
@@ -640,7 +640,8 @@ def _update_dims(self, event=None):
             self._position = (0,) * (ndim - old_ndim) + self._position
 
         self._ndim = ndim
-        self._extent_cache = None
+        if 'extent' in self.__dict__:
+            del self.__dict__['extent']
```

Comment:
```suggestion
            del self.extent
```

(the documented way to invalidate cached_property)

Reply:
changed